### PR TITLE
[12.0][FIX+IMP] website_sale_attribute_filter_price: Refactor code

### PR DESCRIPTION
**Pull request by Tardo**
Now this module its more extensible and configurable:
- Uses Odoo Class
- Allows overwrite slider options with an html data attribute: `data-options`

cc @tecnativa TT23524

### CODE REVIEW

File: website_sale_attribute_filter_price/static/description/index.html [hunk 1]
```diff
@@ -3,7 +3,7 @@
 <html xmlns="http://www.w3.org/1999/xhtml" xml:lang="en" lang="en">
 <head>
 <meta http-equiv="Content-Type" content="text/html; charset=utf-8" />
-<meta name="generator" content="Docutils 0.15.1: http://docutils.sourceforge.net/" />
+<meta name="generator" content="Docutils 0.16: http://docutils.sourceforge.net/" />
```
Comment (pedrobaeza) [inline website_sale_attribute_filter_price/static/description/index.html]: Don't change this. Create a virtualenv or similar where you keep 0.15.1 version for local running of OCA readme generator

**Review by pedrobaeza — APPROVED** [review]

**Review by chienandalu — APPROVED** [review]
:+1: Functional ok

**Comment by Tardo** [comment]
@pedrobaeza updated

**Pull request merged**